### PR TITLE
feat(dashboards): metric widget modal edit

### LINKED ISSUE #64942: Embed the metrics page querying experience into a preview/modal/tray

### PR DESCRIPTION
**Pull request by obostjancic**
- closes #64942

image

removal of inline editor and widget builder option will be done in followup PRs

### CODE REVIEW

File: static/app/actionCreators/modal.tsx [hunk 1]
```diff
@@ -332,7 +333,13 @@ export async function openWidgetViewerModal({
...
   ...options
 }: WidgetViewerModalOptions & {onClose?: () => void}) {
-  const mod = await import('sentry/components/modals/widgetViewerModal');
+  const getModal = () => {
+    if (options.widget.widgetType === WidgetType.METRICS) {
+      return import('sentry/components/modals/metricWidgetViewerModal');
+    }
+    return import('sentry/components/modals/widgetViewerModal');
+  };
+  const mod = await getModal();
```
Comment (ArthurKnaus) [inline static/app/actionCreators/modal.tsx:342]: No hard opinion here, just my what I would do 😄 
```suggestion
  const modalPromise = options.widget.widgetType === WidgetType.METRICS
    ? import('sentry/components/modals/metricWidgetViewerModal')
    : import('sentry/components/modals/widgetViewerModal');
  
  const mod = await modalPromise;
```

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    widget,
+    selection,
+  ]);
+
+  return (
+    <Fragment>
+      <OrganizationContext.Provider value={organization}>
+        <Header closeButton>
+          <WidgetHeader>
+            {/* { TODO: extract to a separate component in a followup } */}
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:129]: hehe, already wanted to suggest this 😁

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+                    widgets: [metricWidgetQueryParams],
+                    project: selection.projects,
+                    ...selection.datetime,
+                    environment: selection.environments,
+                  })
+                );
+              }}
+            >
+              {t('Open in Metrics')}
+            </Button>
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:214]: `<LinkButton />` ?

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+            <Button
+              onClick={() => {
+                closeModal();
+              }}
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:218]: Does anything prevent us from doing this?
```suggestion
              onClick={closeModal}
```

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            >
+              {t('Close')}
+            </Button>
+            <Button
+              color="primary"
+              priority="primary"
+              onClick={() => {
+                handleSubmit();
+                closeModal();
+              }}
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:228]: If you add `closeModal()` to `handleSubmit`, you could simply do 
```suggestion
              onClick={handleSubmit}
```

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+      leadingItems: [<IconSiren key="icon" />],
+      key: 'add-alert',
+      label: t('Create Alert'),
+      disabled: !createAlert,
+      onAction: () => {
+        trackAnalytics('ddm.create-alert', {
+          organization,
+          source: 'widget',
+        });
+        Sentry.metrics.increment('ddm.dashboards.widget.alert');
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:263]: imho we should not add the ddm analytics events to dashboard.
I guess, we have the same problem when it comes to the query builder 🤔

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+    const settingsItem = {
+      leadingItems: [<IconSettings key="icon" />],
+      key: 'settings',
+      label: t('Metric Settings'),
+      disabled: !customMetric,
+      onAction: () => {
+        trackAnalytics('ddm.widget.settings', {
+          organization,
+        });
+        Sentry.metrics.increment('ddm.dashboards.widget.settings');
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:276]: tracking

File: static/app/components/modals/metricWidgetViewerModal.tsx [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+        size: 'md',
+        showChevron: false,
+        icon: <IconEllipsis direction="down" size="sm" />,
+      }}
+      position="bottom-end"
+    />
+  );
+}
+
+export default withPageFilters(MetricWidgetViewerModal);
```
Comment (ArthurKnaus) [inline static/app/components/modals/metricWidgetViewerModal.tsx:301]: why not use `usePageFilters`? copy pasta? 🙃

**Review by ArthurKnaus — APPROVED** [review]

File: static/app/components/modals/metricWidgetViewerModal.tsx (hunk 8 above)
Comment (obostjancic) [reply static/app/components/modals/metricWidgetViewerModal.tsx]: Yep

File: static/app/views/dashboards/widgetCard/metricWidgetCard/index.tsx [hunk 9]
```diff
@@ -287,6 +282,19 @@ export function MetricWidgetChartContainer({
...
+
+  return {
+    mri: parsed.mri,
+    op: parsed.op,
+    query: query.conditions,
+    groupBy: query.columns,
+    displayType: toMetricDisplayType(widget.displayType),
+  };
+}
+
```
[the remark this answers is not in the document]
Comment (obostjancic) [reply static/app/views/dashboards/widgetCard/metricWidgetCard/index.tsx:297]: sure!

File: static/app/actionCreators/modal.tsx (hunk 1 above)
Comment (obostjancic) [reply static/app/actionCreators/modal.tsx]: Yep, its a bit more compact :D

File: static/app/components/modals/metricWidgetViewerModal.tsx (hunk 4 above)
Comment (obostjancic) [reply static/app/components/modals/metricWidgetViewerModal.tsx]: absolutely nothing!

**Pull request merged**